src/reflect/scala/reflect/internal/pickling/UnPickler.scala

```diff
@@ -701,14 +703,14 @@ abstract class UnPickler {
       *
       * Useful for reporting on stub errors and cyclic errors.
       */
-    private var completingStack = List.empty[Symbol]
+    private var completingStack = new mutable.ArrayBuffer[Symbol](128)
```

Comment:
When compiling the compiler, no instance of this stack appears to grow beyond length 1. In a REPL session I found a case where the stack had two elements. So 128, the initial size below of the `ArrayBuffer`, seems too large.

I was somewhat surprised to see that we maintain a separate stack for each `Unpickler.Scan`, rather than having a global stack. Could you comment on why that is?

Reply:
> I was somewhat surprised to see that we maintain a separate stack for each Unpickler.Scan, rather than having a global stack. Could you comment on why that is?
  
This is an oversight... Last commit makes it global.